### DISCUSSION TITLE
Elasticsearch failing after a few days

### DISCUSSION THREAD

**Discussion by Sazerak** [Security-Onion-Solutions/securityonion#7810] [Unsupported Versions]
Hoping for some help troubleshooting this. I changed hardware recently and also upgraded to the latest version of SO2 and recently my elasticsearch container has been constantly going into missing or error after a few days. I'm guessing it's probably related to my new setup, but I can't say for sure and I'm not sure how to troubleshoot this.

I've looked at the logs in /opt/so/log/elasticsearch but nothing really stood out to me. I've tried running so-elasticsearch-restart but that usually just hangs so the only way I've found to bring elasticsearch back is to reboot the system. Then usually within 5-7 days it'll fail again.

**Comment by dougburks** [comment]
What are the specs of your system?
https://docs.securityonion.net/en/2.3/hardware.html#minimum-specs

**Reply by Sazerak** [reply]
I have 4 vCPUs, 16GB RAM, 150GB disk space for / and 1TB for /nsm.

**Reply by dougburks** [reply]
How much log volume are you ingesting into Elasticsearch? Depending on log volume, 16GB RAM may not be enough.

> I've tried running so-elasticsearch-restart but that usually just hangs

Is there any output at all when this happens?

Have you checked `sudo docker logs so-elasticsearch` for additional clues?

**Reply by Sazerak** [reply]
> How much log volume are you ingesting into Elasticsearch? Depending on log volume, 16GB RAM may not be enough.

I hadn't thought of that. Where would I check this? I don't think I specified an amount other than the default.

> I've tried running so-elasticsearch-restart but that usually just hangs
> 
> Is there any output at all when this happens?

I'd have to try again the next time the problem happens (which should be in the next couple of days), but as far as I remember, there was no output.

> Have you checked `sudo docker logs so-elasticsearch` for additional clues?

I just ran this command and got an error with JRE. The output is below. Could this be the cause?

```Importing PKCS12 keypair into Java keystore
Importing keystore /usr/share/elasticsearch/config/elasticsearch.p12 to /usr/share/elasticsearch/config/sokeys...
Entry for alias 1 successfully imported.
Import command completed:  1 entries successfully imported, 0 entries failed or cancelled
Exception in thread "main" java.lang.RuntimeException: starting java failed with [1]
output:
#
# There is insufficient memory for the Java Runtime Environment to continue.
# Native memory allocation (mmap) failed to map 5687476224 bytes for committing reserved memory.
# An error report file with more information is saved as:
# logs/hs_err_pid196.log
error:
OpenJDK 64-Bit Server VM warning: INFO: os::commit_memory(0x000000066d000000, 5687476224, 0) failed; error='Not enough space' (errno=12)
        at org.elasticsearch.tools.launchers.JvmOption.flagsFinal(JvmOption.java:119)
        at org.elasticsearch.tools.launchers.JvmOption.findFinalOptions(JvmOption.java:81)
        at org.elasticsearch.tools.launchers.JvmErgonomics.choose(JvmErgonomics.java:38)
        at org.elasticsearch.tools.launchers.JvmOptionsParser.jvmOptions(JvmOptionsParser.java:135)
        at org.elasticsearch.tools.launchers.JvmOptionsParser.main(JvmOptionsParser.java:86)
Importing PKCS12 keypair into Java keystore
Importing keystore /usr/share/elasticsearch/config/elasticsearch.p12 to /usr/share/elasticsearch/config/sokeys...
Warning: Overwriting existing alias 1 in destination keystore
Entry for alias 1 successfully imported.
Import command completed:  1 entries successfully imported, 0 entries failed or cancelled
Importing PKCS12 keypair into Java keystore
Importing keystore /usr/share/elasticsearch/config/elasticsearch.p12 to /usr/share/elasticsearch/config/sokeys...
Warning: Overwriting existing alias 1 in destination keystore
Entry for alias 1 successfully imported.
Import command completed:  1 entries successfully imported, 0 entries failed or cancelled```

**Reply by dougburks** [reply]
> I hadn't thought of that. Where would I check this? I don't think I specified an amount other than the default.

https://docs.securityonion.net/en/2.3/hardware.html#standalone-deployments
https://docs.securityonion.net/en/2.3/elasticsearch.html#heap-size

> I just ran this command and got an error with JRE. The output is below. Could this be the cause?

Yes, looks like a memory issue:
> There is insufficient memory for the Java Runtime Environment to continue.
> Native memory allocation (mmap) failed to map 5687476224 bytes for committing reserved memory.

**Reply by Sazerak** [reply]
Thanks for the insight. I tried changing the esheap up to roughly 7GB (vs the ~5.5GB default for my system) as per [https://github.com/Security-Onion-Solutions/securityonion/discussions/3932](url) and I will see if that helps.

**Reply by dougburks** [reply]
If it was previously failing to map 5.5GB then I would suspect that you would need to either decrease esheap or increase your overall RAM.
